sdk/samples/iot/hub/src/sample_pnp_thermostat_component.c

```diff
@@ -0,0 +1,363 @@
+    {
+      printf("Could not get reported property: error code = 0x%08x\n", result);
+      return false;
+    }
+    else if (
+        (result = az_iot_hub_client_twin_patch_get_publish_topic(
+             client, get_request_id(), mqtt_message->topic, mqtt_message->topic_length, NULL))
+        != AZ_OK)
+    {
+      printf("Error to get reported property topic with status: error code = 0x%08x\n", result);
+      return false;
+    }
+
+    handle->send_max_temp_property = false;
+
+    return true;
+  }
+  else
+  {
+    return false;
+  }
+}
+
+az_result sample_pnp_thermostat_get_telemetry_message(
+    az_iot_hub_client* client,
+    sample_pnp_thermostat_component* handle,
+    sample_pnp_mqtt_message* mqtt_message)
+{
+  az_result result;
+  if (az_failed(
+          result = pnp_helper_get_telemetry_topic(
+              client, NULL, handle->component_name, mqtt_message->topic, mqtt_message->topic_length, NULL)))
+  {
+    return result;
+  }
+
+  if (az_failed(
+          result = build_telemetry_message(
+              handle, mqtt_message->payload_span, &mqtt_message->out_payload_span)))
+  {
+    printf("Could not build telemetry payload: error code = 0x%08x\n", result);
+    return result;
+  }
+
+  return result;
+}
+
+az_result sample_pnp_thermostat_process_property_update(
+    az_iot_hub_client* client,
+    sample_pnp_thermostat_component* handle,
+    az_span component_name,
+    az_span property_name,
+    az_json_token* property_value,
+    int32_t version,
+    sample_pnp_mqtt_message* mqtt_message)
+{
+  az_result result;
+
+  if (!az_span_is_content_equal(handle->component_name, component_name))
+  {
+    return AZ_ERROR_ITEM_NOT_FOUND;
+  }
+
+  if (!az_span_is_content_equal(desired_temp_property_name, property_name))
+  {
+    printf(
+        "PnP property=%.*s is not supported on thermostat component\r\n",
+        az_span_size(property_name),
+        az_span_ptr(property_name));
+  }
+
+  double parsed_value = 0;
+  if (az_failed(az_json_token_get_double(property_value, &parsed_value)))
+  {
+    result = pnp_helper_create_reported_property_with_status(
+        mqtt_message->payload_span,
+        component_name,
+        property_name,
+        append_double,
+        (void*)&parsed_value,
+        401,
+        version,
+        temp_response_description_failed,
+        &mqtt_message->out_payload_span);
+  }
+  else
+  {
+    handle->current_temperature = parsed_value;
+    if (handle->current_temperature > handle->max_temperature)
+    {
+      handle->max_temperature = handle->current_temperature;
+      handle->send_max_temp_property = true;
+    }
+
+    if (handle->current_temperature < handle->min_temperature)
+    {
+      handle->min_temperature = handle->current_temperature;
+    }
+
+    /* Increment the avg count, add the new temp to the total, and calculate the new avg */
+    handle->device_temperature_avg_count++;
+    handle->device_temperature_avg_total += handle->current_temperature;
```

Comment:
possible overflow?

Reply:
In theory yes but for the purposes of, unlikely.